File: app/src/main/java/com/controlmoblie/resolver/AppResolver.java
package com.controlmoblie.resolver;

import java.util.Map;
import java.util.HashMap;

public class AppResolver {
    private static final Map<String, String> ALIASES = new HashMap<>();

    static {
        // ===== 系统应用 =====
        ALIASES.put("设置", "com.android.settings");
        ALIASES.put("相机", "com.android.camera");
        ALIASES.put("日历", "com.android.calendar");
        ALIASES.put("时钟", "com.android.deskclock");
        ALIASES.put("闹钟", "com.android.deskclock");
        ALIASES.put("计算器", "com.miui.calculator");
        ALIASES.put("天气", "com.miui.weather2");
        ALIASES.put("相册", "com.miui.gallery");
        ALIASES.put("文件", "com.android.fileexplorer");
        ALIASES.put("浏览器", "com.android.browser");
        ALIASES.put("小米浏览器", "com.miui.browser");
        ALIASES.put("电话", "com.android.dialer");
        ALIASES.put("拨号", "com.android.dialer");
        ALIASES.put("短信", "com.android.mms");
        ALIASES.put("联系人", "com.android.contacts");
        ALIASES.put("邮件", "com.android.email");
        ALIASES.put("邮箱", "com.android.email");
        ALIASES.put("下载", "com.android.providers.downloads.ui");
        ALIASES.put("录音机", "com.android.soundrecorder");
        ALIASES.put("应用商店", "com.xiaomi.market");
        ALIASES.put("米家", "com.xiaomi.smarthome");
        ALIASES.put("扫一扫", "com.xiaomi.scanner");
        ALIASES.put("扫", "com.xiaomi.scanner");
        ALIASES.put("便签", "com.miui.notes");
        ALIASES.put("笔记", "com.miui.notes");
        ALIASES.put("指南针", "com.miui.compass");
        ALIASES.put("手机管家", "com.miui.cleanmaster");
        ALIASES.put("屏幕录制", "com.miui.screenrecorder");
        ALIASES.put("主题商店", "com.miui.themestore");
        ALIASES.put("小米换机", "com.miui.huanji");
        ALIASES.put("小米相册编辑", "com.miui.mediaeditor");
        ALIASES.put("小米云盘", "com.miui.newmidrive");
        ALIASES.put("小米画报", "com.mfashiongallery.emag");
        ALIASES.put("小米运动健康", "com.mi.health");
        ALIASES.put("万能遥控", "com.duokan.phone.remotecontroller");
        ALIASES.put("讯飞输入法", "com.iflytek.inputmethod.miui");
        ALIASES.put("小爱同学", "com.xiaomi.mibrain.speech");

        // ===== 腾讯系 =====
        ALIASES.put("微信", "com.tencent.mm");
        ALIASES.put("qq", "com.tencent.mobileqq");
        ALIASES.put("qq音乐", "com.tencent.qqmusic");
        ALIASES.put("qq邮箱", "com.tencent.androidqqmail");
        ALIASES.put("qq浏览器", "com.tencent.mtt");
        ALIASES.put("腾讯会议", "com.tencent.wemeet.app");
        ALIASES.put("微信读书", "com.tencent.weread");
        ALIASES.put("企业微信", "com.tencent.wework");
        ALIASES.put("企业号", "com.tencent.wework");

        // ===== 阿里系 =====
        ALIASES.put("支付宝", "com.eg.android.AlipayGphone");
        ALIASES.put("淘宝", "com.taobao.taobao");
        ALIASES.put("钉钉", "com.alibaba.android.rimet");
        ALIASES.put("闲鱼", "com.taobao.idlefish");
        ALIASES.put("菜鸟", "com.cainiao.wireless");

        // ===== 字节系 =====
        ALIASES.put("抖音", "com.ss.android.ugc.aweme");
        ALIASES.put("今日头条", "com.ss.android.article.news");
        ALIASES.put("新闻", "com.ss.android.article.news");
        ALIASES.put("飞书", "com.ss.android.lark");

        // ===== 百度系 =====
        ALIASES.put("百度", "com.baidu.searchbox");
        ALIASES.put("百度地图", "com.baidu.BaiduMap");
        ALIASES.put("百度网盘", "com.baidu.netdisk");

        // ===== 网易系 =====
        ALIASES.put("网易云音乐", "com.netease.cloudmusic");
        ALIASES.put("音乐", "com.netease.cloudmusic");
        ALIASES.put("网易有道词典", "com.youdao.dict");

        // ===== 视频/娱乐 =====
        ALIASES.put("哔哩哔哩", "tv.danmaku.bili");
        ALIASES.put("b站", "tv.danmaku.bili");
        ALIASES.put("bili", "tv.danmaku.bili");
        ALIASES.put("快手", "com.smile.gifmaker");
        ALIASES.put("优酷", "com.youku.phone");
        ALIASES.put("小红书", "com.xingin.xhs");
        ALIASES.put("微博", "com.sina.weibo");
        ALIASES.put("即刻", "com.ruguoapp.jike");
        ALIASES.put("豆瓣", "com.douban.frodo");
        ALIASES.put("看理想", "com.kanlixiang.android");
        ALIASES.put("西西弗", "com.sisyphe.mobile");

        // ===== 出行 =====
        ALIASES.put("高德地图", "com.autonavi.minimap");
        ALIASES.put("地图", "com.autonavi.minimap");
        ALIASES.put("滴滴", "com.sdu.didi.psnger");
        ALIASES.put("打车", "com.sdu.didi.psnger");
        ALIASES.put("美团", "com.sankuai.meituan");
        ALIASES.put("外卖", "com.sankuai.meituan");
        ALIASES.put("饿了么", "me.ele");
        ALIASES.put("12306", "com.MobileTicket");
        ALIASES.put("火车票", "com.MobileTicket");
        ALIASES.put("铁路", "com.MobileTicket");
        ALIASES.put("交管12123", "com.tmri.app.main");
        ALIASES.put("日产智联", "com.szlanyou.nissaniov");

        // ===== 购物 =====
        ALIASES.put("京东", "com.jingdong.app.mall");
        ALIASES.put("拼多多", "com.xunmeng.pinduoduo");

        // ===== 金融 =====
        ALIASES.put("招商银行", "cmb.pb");
        ALIASES.put("工商银行", "com.icbc");
        ALIASES.put("中信银行", "com.ecitic.bank.mobile");
        ALIASES.put("个人所得税", "cn.gov.tax.its");

        // ===== 工具/效率 =====
        ALIASES.put("知乎", "com.zhihu.android");
        ALIASES.put("WPS", "cn.wps.moffice_eng");
        ALIASES.put("wps", "cn.wps.moffice_eng");
        ALIASES.put("向日葵", "com.oray.sunlogin");
        ALIASES.put("夸克扫描", "com.quark.scanking");
        ALIASES.put("V2RayNG", "com.v2ray.ang");
        ALIASES.put("DeepSeek", "com.example.deepseekchat");
        ALIASES.put("ChatGPT", "com.openai.chatgpt");
        ALIASES.put("GitHub", "com.github.android");
        ALIASES.put("Bing", "com.microsoft.bing");
        ALIASES.put("CSDN", "net.csdn.csdnplus");
        ALIASES.put("力扣", "com.lingkou.leetcode");
        ALIASES.put("Coursera", "org.coursera.android");
        ALIASES.put("维基百科", "org.wikipedia");
        ALIASES.put("YouTube", "com.google.android.youtube");
        ALIASES.put("Instagram", "com.instagram.android");
        ALIASES.put("Threads", "com.instagram.barcelona");

        // ===== 招聘 =====
        ALIASES.put("Boss直聘", "com.hpbr.bosszhipin");

        // ===== 运营商 =====
        ALIASES.put("中国联通", "com.sinovatech.unicom.ui");
    }

    public static String resolve(String name) {
        String result = ALIASES.get(name);
        return result != null ? result : name;
    }
}

File: app/src/main/java/com/controlmoblie/util/AppResolver.kt
package com.controlmoblie.util

object AppResolver {

    private val aliases = mapOf(
        // ===== 系统应用 =====
        "设置" to "com.android.settings",
        "相机" to "com.android.camera",
        "日历" to "com.android.calendar",
        "时钟" to "com.android.deskclock",
        "闹钟" to "com.android.deskclock",
        "计算器" to "com.miui.calculator",
        "天气" to "com.miui.weather2",
        "相册" to "com.miui.gallery",
        "文件" to "com.android.fileexplorer",
        "浏览器" to "com.android.browser",
        "小米浏览器" to "com.miui.browser",
        "电话" to "com.android.dialer",
        "拨号" to "com.android.dialer",
        "短信" to "com.android.mms",
        "联系人" to "com.android.contacts",
        "邮件" to "com.android.email",
        "邮箱" to "com.android.email",
        "下载" to "com.android.providers.downloads.ui",
        "录音机" to "com.android.soundrecorder",
        "应用商店" to "com.xiaomi.market",
        "米家" to "com.xiaomi.smarthome",
        "扫一扫" to "com.xiaomi.scanner",
        "扫" to "com.xiaomi.scanner",
        "便签" to "com.miui.notes",
        "笔记" to "com.miui.notes",
        "指南针" to "com.miui.compass",
        "手机管家" to "com.miui.cleanmaster",
        "屏幕录制" to "com.miui.screenrecorder",
        "主题商店" to "com.miui.themestore",
        "小米换机" to "com.miui.huanji",
        "小米相册编辑" to "com.miui.mediaeditor",
        "小米云盘" to "com.miui.newmidrive",
        "小米画报" to "com.mfashiongallery.emag",
        "小米运动健康" to "com.mi.health",
        "万能遥控" to "com.duokan.phone.remotecontroller",
        "讯飞输入法" to "com.iflytek.inputmethod.miui",
        "小爱同学" to "com.xiaomi.mibrain.speech",

        // ===== 腾讯系 =====
        "微信" to "com.tencent.mm",
        "qq" to "com.tencent.mobileqq",
        "qq音乐" to "com.tencent.qqmusic",
        "qq邮箱" to "com.tencent.androidqqmail",
        "qq浏览器" to "com.tencent.mtt",
        "腾讯会议" to "com.tencent.wemeet.app",
        "微信读书" to "com.tencent.weread",
        "企业微信" to "com.tencent.wework",
        "企业号" to "com.tencent.wework",

        // ===== 阿里系 =====
        "支付宝" to "com.eg.android.AlipayGphone",
        "淘宝" to "com.taobao.taobao",
        "钉钉" to "com.alibaba.android.rimet",
        "闲鱼" to "com.taobao.idlefish",
        "菜鸟" to "com.cainiao.wireless",

        // ===== 字节系 =====
        "抖音" to "com.ss.android.ugc.aweme",
        "今日头条" to "com.ss.android.article.news",
        "新闻" to "com.ss.android.article.news",
        "飞书" to "com.ss.android.lark",

        // ===== 百度系 =====
        "百度" to "com.baidu.searchbox",
        "百度地图" to "com.baidu.BaiduMap",
        "百度网盘" to "com.baidu.netdisk",

        // ===== 网易系 =====
        "网易云音乐" to "com.netease.cloudmusic",
        "音乐" to "com.netease.cloudmusic",
        "网易有道词典" to "com.youdao.dict",

        // ===== 视频/娱乐 =====
        "哔哩哔哩" to "tv.danmaku.bili",
        "b站" to "tv.danmaku.bili",
        "bili" to "tv.danmaku.bili",
        "快手" to "com.smile.gifmaker",
        "优酷" to "com.youku.phone",
        "小红书" to "com.xingin.xhs",
        "微博" to "com.sina.weibo",
        "即刻" to "com.ruguoapp.jike",
        "豆瓣" to "com.douban.frodo",
        "看理想" to "com.kanlixiang.android",
        "西西弗" to "com.sisyphe.mobile",

        // ===== 出行 =====
        "高德地图" to "com.autonavi.minimap",
        "地图" to "com.autonavi.minimap",
        "滴滴" to "com.sdu.didi.psnger",
        "打车" to "com.sdu.didi.psnger",
        "美团" to "com.sankuai.meituan",
        "外卖" to "com.sankuai.meituan",
        "饿了么" to "me.ele",
        "12306" to "com.MobileTicket",
        "火车票" to "com.MobileTicket",
        "铁路" to "com.MobileTicket",
        "交管12123" to "com.tmri.app.main",
        "日产智联" to "com.szlanyou.nissaniov",

        // ===== 购物 =====
        "京东" to "com.jingdong.app.mall",
        "拼多多" to "com.xunmeng.pinduoduo",

        // ===== 金融 =====
        "招商银行" to "cmb.pb",
        "工商银行" to "com.icbc",
        "中信银行" to "com.ecitic.bank.mobile",
        "个人所得税" to "cn.gov.tax.its",

        // ===== 工具/效率 =====
        "知乎" to "com.zhihu.android",
        "WPS" to "cn.wps.moffice_eng",
        "wps" to "cn.wps.moffice_eng",
        "向日葵" to "com.oray.sunlogin",
        "夸克扫描" to "com.quark.scanking",
        "V2RayNG" to "com.v2ray.ang",
        "DeepSeek" to "com.example.deepseekchat",
        "ChatGPT" to "com.openai.chatgpt",
        "GitHub" to "com.github.android",
        "Bing" to "com.microsoft.bing",
        "CSDN" to "net.csdn.csdnplus",
        "力扣" to "com.lingkou.leetcode",
        "Coursera" to "org.coursera.android",
        "维基百科" to "org.wikipedia",
        "YouTube" to "com.google.android.youtube",
        "Instagram" to "com.instagram.android",
        "Threads" to "com.instagram.barcelona",

        // ===== 招聘 =====
        "Boss直聘" to "com.hpbr.bosszhipin",

        // ===== 运营商 =====
        "中国联通" to "com.sinovatech.unicom.ui",
    )

    fun resolve(name: String): String {
        return aliases[name] ?: name
    }
}
